cosign/README.md

```diff
@@ -0,0 +1,79 @@
+Cosign builds on `sigstore-go` and extends its capabilities to include container image signing, verification, and storage in registries.
+
+Using `sigstore-go` library makes it a more streamlined choice for developers seeking to implement lightweight and flexible signature verification.
+Additionally, the Sigstore team plans to refactor parts of Cosign to rely more heavily on `sigstore-go`, making the interaction between the two libraries even closer.
+
+## Concepts
+
+Artifact Signatures  
+
+- Artifacts may be signed using different cryptographic schemes. The verifier must support a variety of signature formats, including RSA, ECDSA, and Ed25519.
+- The use of DSSE helps bundle signatures with metadata, providing additional context and validation capabilities.
+
+Transparency Logs  
+
+- Transparency logs, such as those maintained by the Sigstore project, provide an immutable record of signature entries.
+- Verifying against these logs adds an extra layer of assurance that a signature has been publicly disclosed and has not been secretly revoked or altered.
+- As a primary and recommanded solution, Cosign will then store the signature and certificate in the Rekor transparency log.
+
+Verification Workflows  
+
+- The verifier must account for different workflows, such as verifying container images, software binaries, or other artifacts.
+- Each workflow may involve different states of metadata, such as embedded signatures versus detached signatures, and requires customization accordingly.
+- `sigstore-go` library includes a few abstractions to support different use cases, testing, and extensibility:
+
+  - `SignedEntity` - an interface type respresenting a signed message or attestation, with a signature and metadata, implemented by `Bundle`, a type which wraps the `Bundle` type from `protobuf-specs`.
+  - `TrustedMaterial` - an interface type representing a trusted set of keys or certificates for verifying certificates, timestamps, and artifact transparency logs, implemented by `TrustedRoot`
+
+## Scenarios
+
+`sigstore-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources.
+These scenarios can be categorized into following main types and for this library we are going to implement the first four verification for basic user scenarios.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifies signature using **OIDC/Fulcio**-based trust (no private key needed). | CI/CD pipelines or automated workflows where no private key management is needed. |
```

Comment:
The term "private key" is not related. What's the trust anchor of this scenario?

Reply:
The primary trust anchor is the Fulcio Root Certificate Authority (CA). Update as following:
```
Fulcio issues short-lived certificates binding an ephemeral key to an OpenID Connect identity. Signing events are logged in Rekor, a signature transparency log.
```